docs/developers/best_practices.md

```diff
@@ -45,19 +45,54 @@ sample for further guidance on using dynamic shapes.
 
 ## Practices for compilation settings
 
-TODO: mention parameters to tune
-
 TODO: which compiler targets to use (try both CUDA and Vulkan?)
 
 TODO: use the most specific LLVM target triple you can?
 
-## Practices for runtime use
+### Tuning compilation heuristics
 
-### Do the minimum amount of work: cache queries and reuse buffers
+IREE runs its own suite of benchmarks continuously using the definitions at
+https://github.com/google/iree/tree/main/benchmarks. The flags set for these
+benchmarks represent the latest manually tuned values for workloads we track
+closely and referencing them may help with your own search for peak performance.
 
-Try to front-load queries, particularly queries using strings that look up into
-maps like `iree_runtime_session_call_by_name`, so that hot sections of code are
-doing the minimum amount of work: routing inputs through buffers, scheduling
-runtime calls, and routing outputs through other buffers.
+Here is a non-exhaustive list of flags which can be tuned when compiling
```

Comment:
I am not sure I want to document these. These are all temporary and will likely get dropped. We dont want people to rely on these flags.

Reply:
Sure, though we already have people asking for settings to tune, and some of the existing flags have large impacts on performance. I'd like to be able to point people in the right direction for both early project evaluation and for stable settings.

How about I remove the specific flags but keep the reference to the benchmarks directory and note about searching the source code? I'll also elaborate on the language based on Lei's comment.

Reply:
Yeah, that sounds good to me. I am working on dropping these flags for good. So would rather people not rely on them